lib/systems/default.nix

```diff
@@ -102,7 +102,7 @@ rec {
       in
         if final.parsed.kernel.name == pkgs.stdenv.hostPlatform.parsed.kernel.name &&
            pkgs.stdenv.hostPlatform.isCompatible final
-        then "${pkgs.runtimeShell} -c"
+        then "${pkgs.runtimeShell} -c '$@' --"
```

Comment:
You might want `"${pkgs.runtimeShell} -c '\"$@\"' --"` actually, but I'm just guessing.